package-examples/ghost/fn/update-blog-host.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: fn.kpt.dev/v1alpha1
+kind: StarlarkRun
+metadata:
+  name: update-blog-host
+params:
+  host: example.com
+source: |
+  host = ctx.resource_list["functionConfig"]["params"]["host"]
+
+  for resource in ctx.resource_list["items"]:
+    if resource["kind"] == "Ingress":
+      resource["spec"]["rules"][0]["host"] = host
+
+    if resource["kind"] == "Deployment":
+      container = resource["spec"]["template"]["spec"]["containers"][0]
+
+      for env in container["env"]:
+        if env["name"] == "GHOST_HOST":
+          env["value"] = host
```

Comment:
OOC, is ApplyReplacements capable of copying the host value from the Ingress resource to this environment variable in the associative list?

Reply:
I will check.

Let's say if ApplyReplacements supports that, then, are you thinking of treating the value of host in `ingress` resource to be source of truth ?

Reply:
the docs says it is possible to target value in a sequence node, so will give it a try.

https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/replacements/#field-path-format